src/textual/widgets/__init__.py

```diff
@@ -1,21 +1,45 @@
-from ._footer import Footer
-from ._header import Header
-from ._button import Button
-from ._placeholder import Placeholder
-from ._static import Static
-from ._tree_control import TreeControl, TreeClick, TreeNode, NodeID
-from ._directory_tree import DirectoryTree, FileClick
-
-__all__ = [
-    "Button",
-    "DirectoryTree",
-    "FileClick",
-    "Footer",
-    "Header",
-    "Placeholder",
-    "Static",
-    "TreeClick",
-    "TreeControl",
-    "TreeNode",
-    "NodeID",
-]
+from __future__ import annotations
+from importlib import import_module
+import typing
+
+if typing.TYPE_CHECKING:
+    from ..widget import Widget
+
+# Let's decrease startup time by lazy loading our Widgets:
+# ⚠ For any new built-in Widget we create, not only we have to add them to the following dictionary, but also to the
+# `__init__.pyi` file in this same folder - otherwise text editors and type checkers won't be able to "see" them.
+_WIDGETS_LAZY_LOADING_MAPPING: dict[str, str] = {
+    "Footer": "._footer:Footer",
```

Comment:
I opted for a simple-and-stupid approach; combined to the fact that the lazy-loaded Widget classes are kept in a (simple) cache this implementation should do the job? :slightly_smiling_face: :crossed_fingers:

Reply:
I think we can eliminate this dict because the imports can al be deduced from the name. i.e. if you know the name `Button`, you can deduce the import is `_button.Button`

I'd suggest using `__all__` to keep a list of the widgets, and deducing the imports from there.

Reply:
My first implementation was using such guesses indeed, rather than having to maintain that mapping; however it doesn't work with the various types Widgets can also carry alongside their main class?
i.e. it's easy to infer that `TreeControl` is the result of `from ._tree_control import TreeControl`, but without such a mapping how do we determine that `NodeID`is the result of `from ._tree_control import NodeID`, since there is no links between the string _"tree_control"_ and the string _"NodeID"_ ?

Reply:
True. But we can adopt this convention going forward. The TreeControl hasn’t been updated from the old API, and we will probably not expose symbols other than the Widgets themselves.

Reply:
Alright, let's do that! And on top of the automatic inference, which simplifier the maintenance of our built-in Widgets, I guess we can also fall back to such a manual mapping if we need to, for Widgets that would require "companion classes"? :slightly_smiling_face:

Reply:
I'm thinking that we can probably mostly avoid companion classes. Small objects can go in the class itself. For instance `TreeControl.NodeID`. Will see how that works out.